### PR TITLE
Consider core dns in the preflight DNS check

### PR DESCRIPTION
**Pull request by pipo02mix**
Since now [coredns](https://github.com/coredns/coredns) it is a real alternative for DNS server in Kubernetes I would like to be able to run sonobuoy without modifying default labels.

### CODE REVIEW

File: pkg/client/preflight.go [hunk 1]
```diff
@@ -54,20 +54,31 @@ const (
 	kubeSystemNamespace = "kube-system"
 	kubeDNSLabelKey     = "k8s-app"
 	kubeDNSLabelValue   = "kube-dns"
+	coreDNSLabelValue   = "coredns"
```
Comment (timothysc) [inline pkg/client/preflight.go]: What deployment tool is this sets this label?  Kubeadm overloads for the kube-dns label w/coreos.

Comment (timothysc) [reply pkg/client/preflight.go]: I think we should check kubespray and make a call, I'll try to see if they do the same.

Comment (pipo02mix) [reply pkg/client/preflight.go]: There is a fresh PR approved for adding coredns to kubespray. There I dont see any overwriting action

https://github.com/kubernetes-incubator/kubespray/pull/2462

Comment (timothysc) [reply pkg/client/preflight.go]: @pipo02mix - It appears they add `k8s-app : coredns` and that seems like a good addition

**Comment by timothysc** [comment]
@pipo02mix - we use DCO signoff checker on our repos.  Could you amend your commit with `--signoff` 

/cc @liztio for review as you wrote this section

**Comment by pipo02mix** [comment]
I have screwed it up doing the sign-off. For some reason, I could not sign only last two commits. I close this in favor of a [new PR](https://github.com/heptio/sonobuoy/pull/366)

**Pull request closed without merging**